### PR TITLE
[IoT] Add support for system-assigned managed identity for IoT Central applications. Remove geographic location option for new applications.

### PR DESCRIPTION
**Pull request by bretmulvey**
**Description**
Change the help text for "az iot central app create" to use 'East US' instead of 'unitedstates', and remove incorrect parameter from the 'region of the resource group' example. The geographic locations are no longer supported for new applications.

Add "--mi-system-assigned" option for "az iot central app create" to create an app with system-assigned managed identity.

Added "az iot central app identity" command group to show/assign/remove system-assigned managed identity to an existing IoT Central application.

**Testing Guide**

**History Notes**
[IoT] `az iot central app create`: Add new parameter `--mi-system-assigned` to support creating an app with system-assigned managed identity
[IoT] `az iot central app identity show/assign/remove`: Add new commands to manage the system-assigned managed identity to an existing IoT Central app

### CODE REVIEW

**Comment by yonzhan** [comment]
IoT

**Review by wangzelin007 — APPROVED** [review]

**Comment by gansvv** [comment]
@wangzelin007 Can you please review once more and help merge it back to the repo. Thanks

**Comment by zhoxing-ms** [comment]
> "az iot central app create" now allows "--identity SystemAssigned" to create an app with system-assigned managed identity.

This seems to be a new feature that users can perceive. Why does the PR title begin with `{}`? Could we consider writing this change into the history notes?
Such as:
```
[IoT] `az iot central app create`: Allow `--identity SystemAssigned` to create an app with system-assigned managed identity.
```

**Comment by zhoxing-ms** [comment]
Incidentally, the design of `--identity SystemAssigned` does not conform to the unified design guideline of CLI. 
Please refer to this doc [managed_identity_command_guideline](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md) to design managed identity related commands and parameters
As for enabling managed identity during resource creation, the `--mi-system-assigned` should be a more standard design

**Comment by wangzelin007** [comment]
> @wangzelin007 Zelin Wang FTE Can you please review once more and help merge it back to the repo. Thanks

@gansvv 
Please check zhoxing-ms's comments.
And fix the CLI Linter errors and CLI style errors, you can run `azdev linter iot` and `azdev style iot` to check these errors on your local environment.

**Comment by bretmulvey** [comment]
> Incidentally, the design of `--identity SystemAssigned` does not conform to the unified design guideline of CLI. Please refer to this doc [managed_identity_command_guideline](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md) to design managed identity related commands and parameters. As for enabling managed identity during resource creation, the `--mi-system-assigned` should be a more standard design

will fix

**Comment by zhoxing-ms** [comment]
@bretmulvey Any update?  Please note that if you can't address these comments and CI issues today, this PR will not catch up with the release date of this sprint and it can only be postponed to the next sprint (**2022-01-04**)

**Comment by bretmulvey** [comment]
> @bretmulvey Any update? Please note that if you can't address these comments and CI issues today, this PR will not catch up with the release date of this sprint and it can only be postponed to the next sprint (**2022-01-04**)

Understood. I'm pushing for today as early as possible.

**Comment by bretmulvey** [comment]
> @bretmulvey Any update? Please note that if you can't address these comments and CI issues today, this PR will not catch up with the release date of this sprint and it can only be postponed to the next sprint (**2022-01-04**)

@zhoxing-ms I address standards, linting, and style issues. Can create app using --mi-system-assigned identity. Will be added new "identity" command in next sprint so for now we will depend on existing "--set" functionality to update managed identity.

I'm getting a lot of errors from `azdev test test_iot_central_app --live --discover` that I don't know what to do with and I posted question to Azure CLI Partners teams channel.

**Comment by zhoxing-ms** [comment]
except CannotOverwriteExistingCassetteException as ex:
>           raise AssertionError(ex)
E           AssertionError: Can't overwrite existing cassette ('/home/vsts/work/1/s/src/azure-cli/azure/cli/command_modules/iot/tests/latest/recordings/test_iot_central_app.yaml') in your current record mode ('once').
E           No match for the request (<Request (PUT) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.IoTCentral/iotApps/iotc-cli-test000002?api-version=2021-06-01>) was found.
E           Found 1 similar requests with 1 different matcher(s) :
E           
E           1 - (<Request (PUT) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.IoTCentral/iotApps/iotc-cli-test000002?api-version=2018-09-01>).
E           Matchers succeeded : ['method', 'scheme', 'host', 'port', 'path']
E           Matchers failed :
E           _custom_request_query_matcher - assertion failure :
E           None

The CI issue is due to the package `azure-mgmt-iotcentral` does not support multi api-version. Since you have updated the package version, it will use the api-version in the new package, so the api-version will be upgraded from `2018-09-01` to `2021-06-01`

**Comment by zhoxing-ms** [comment]
By the way, could you add some tests for this new parameter?

**Comment by bretmulvey** [comment]
> By the way, could you add some tests for this new parameter?

I have a new test but haven't been able to verify it in the official way because of the pywin32 error (still unresolved).

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -83,6 +83,11 @@ Release History
 
 * GA `az eventgrid system-topic` (#20531)
 
+**IoT Central**
+
+* `az iot central app create --mi-system-assigned`: New `--mi-system-assigned` parameter for app create to assign a system-assigned managed identity to the app (#20448)
+
+
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:90]: Please do not add the `HISTORY.rst` manually. The history notes will be automatically generated by PR title and description. 
Please refer to this doc [submitting-pull-requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests) for details

Comment (bretmulvey) [reply src/azure-cli/HISTORY.rst]: Fixed. Updated title and description.

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -5216,4 +5216,4 @@ azure-cli-vm
 0.1.0b11 (2016-12-12)
 +++++++++++++++++++++
 
-* Preview release.
+* Preview release.
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:5219]: Why there is a change here? Could we not modify it?

**Comment by zhoxing-ms** [comment]
@bretmulvey I modified the PR description to follow the standard format from [submitting-pull-requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests) Please confirm whether the **History Notes** of my modification meets your expectations

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (bretmulvey) [reply src/azure-cli/HISTORY.rst]: fixed

**Review by zhoxing-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**